hyperspy/signal.py

```diff
@@ -3111,6 +3112,79 @@ def crop(self, axis, start=None, end=None, convert_units=False):
         if convert_units:
             self.axes_manager.convert_units(axis)
 
+    def interpolate_on_axis(self,
+                            new_axis,
+                            index_in_axes_manager=0,
+                            inplace=False,
+                            set_navigate=True,
+                            degree=1):
+        """Replaces the axis specified by index_in_axes_manager with the provided new_axis
+        and interpolates data accordingly.
+
+        Parameters
+        ----------
+        new_axis : UniformDataAxis, DataAxis or FunctionalDataAxis
+            Axis which replaces the axis specified with replace_index.
+            If this new axis exceeds the range of the old axis,
+            a warning is raised that the data will be extrapolated.
+
+        index_in_axes_manager : int, default=0
```

Comment:
Rename to `axis` or similar and add support for using axis name when a string is passed, as this is usually done elsewhere in hyperspy?

Reply:
Did that.